core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -44,9 +52,12 @@ private[spark] class CoarseGrainedExecutorBackend(
     hostname: String,
     cores: Int,
     userClassPath: Seq[URL],
-    env: SparkEnv)
+    env: SparkEnv,
+    resourcesFile: Option[String])
```

Comment:
Is it correct to say this is only used in standalone mode? this is how the master tells executors what resource it can use? if that's right, how would this work for other resource managers -- simply, in some other RM-specific way?

Reply:
Yes, the intention is for this to be used by standalone mode.  It has no support for containers or isolation like the other resource managers have so it will pass in a file specifying which resources for that container to use. In the GPU case if the node has 4 gpu's and the executor asked for one, it would pass in one GPU resource and the id of that 1 gpu it should use.  There is a separate jira to add that to standalone mode.
The other resources managers I believe all support some sort of container and isolation such that when the executor starts up it can just use the discoveryScript  to auto discover what resources it has.  We have tested on kubernetes and yarn, I don't have a mesos setup to test on but someone I believe volunteered to work on that side.
You can obviously run other resources manager like YARN without docker and isolation and in that case this wouldn't work, but YARN doesn't give you any other option.  It doesn't tell you what GPU you received for instance, so in the documentation we will put the limitations and required that yarn/kubernetes, etc must be configured in such a way for this to work.  If you are running YARN without isolation then you could also come up with discovery script to make this work, similar to the hacks users are doing now on YARN with gpu scheduling.